translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -5934,6 +5934,38 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
     return mapValue(BV, FNeg);
   }
 
+  case OpFConvert: {
+    SPIRVUnary *BC = static_cast<SPIRVUnary *>(BV);
+    Value *Val = transValue(BC->getOperand(0), F, BB);
+    Type *DestTy = transType(BC->getType());
+    if (Val->getType()->getScalarType()->getPrimitiveSizeInBits() <=
+        DestTy->getScalarType()->getPrimitiveSizeInBits())
+      return mapValue(BV, getBuilder()->CreateFPExt(Val, DestTy));
+
+    ConstrainedFPIntrinsic::RoundingMode RM = ConstrainedFPIntrinsic::rmDynamic;
+    SPIRVFPRoundingModeKind Rounding;
+    if (BC->hasFPRoundingMode(&Rounding)) {
+      switch (Rounding) {
+      case FPRoundingModeRTE:
+        RM = ConstrainedFPIntrinsic::rmToNearest;
+        break;
+      case FPRoundingModeRTZ:
+        RM = ConstrainedFPIntrinsic::rmTowardZero;
+        break;
+      case FPRoundingModeRTP:
+        RM = ConstrainedFPIntrinsic::rmUpward;
+        break;
+      case FPRoundingModeRTN:
+        RM = ConstrainedFPIntrinsic::rmDownward;
+        break;
+      default:
+        llvm_unreachable("");
+      }
+      return mapValue(BV, getBuilder()->CreateFPTruncWithRounding(Val, DestTy, RM));
+    }
+    return mapValue(BV, getBuilder()->CreateFPTrunc(Val, DestTy));
```

Comment:
please use CreateFPTruncWithRounding too if BC->hasFPRoundingMode(&Rounding) == false. I think you can choose any rounding mode in this mode.

Reply:
I think default rounding mode is RTE.

Reply:
OK, but I think it did just do an FPTrunc before.

Reply:
I just checked, and it definitely did do FPTrunc before. So I will leave it like that.